content/tutorials/WebRTC-file-transfers-with-Ably.textile

```diff
@@ -0,0 +1,509 @@
+
+
+
+<a href="/images/tutorials/users-online.png" target="_blank">
+  <img src="/images/tutorials/users-online.png" style="width: 50%" alt="Demo image">
+</a>
+
+<a href="/images/tutorials/users-offline.png" target="_blank">
+  <img src="/images/tutorials/users-offline.png" style="width: 50%" alt="Demo image">
+</a>
+
+h2.
+  default: Step 6 – Sending files
+
+We have been able to identify who is online and ready to receive instant files using WebRTC data channels, now we move to the part where we send files. Add the following to your @ably-datachannel.js@ file.
+
+```[javascript]
+    function PrepareChat(client_id) {
+        if (client_id === clientId) return
+            // Create a new connection
+        currentChat = client_id
+        if (!connections[client_id]) {
+            connections[client_id] = new Connection(client_id, AblyRealtime, true)
+        }
+        document.getElementById('chat').style.display = 'block'
+        render()
+    }
+    
+    
+    AblyRealtime.subscribe(`rtc-signal/${clientId}`, msg => {
+        if (!connections[msg.data.user]) {
+            connections[msg.data.user] = new Connection(msg.data.user, AblyRealtime, false)
+        }
+        connections[msg.data.user].handleSignal(msg.data.signal)
+    })
+    
+    function sendMessage() {
+        if (!messageList[currentChat]) {
+            messageList[currentChat] = {}
+        }
+        var MAX_FSIZE = 2.0
+        var fileInput = document.getElementById('file')
+        var file = fileInput.files[0];
+        var fileReader = new FileReader();
+        if (file) {
+            var mbSize = file.size / (1024 * 1024);
+            if (mbSize > MAX_FSIZE) {
+                alert("Your file is too big, sorry.");
+                // Reset file input
+                return;
+            }
+            fileReader.onload = (e) => { onReadAsDataURL(e, null, file) }
+            fileReader.readAsDataURL(file)
+            render()
+        }
+    }
+```
+"See this step in Github":https://github.com/samuelayo/tutorials/commit/23b5939f13f4e3249b73f074bdfa2643a573259e
+
+**What is going on in the code block above?**
+To share files with a user/member, we need to click the @chat now@ button next to their name. This button, in turn, calls the @prepareChat@ method, which we have just defined.
+
+The @prepareChat@ method receives a single argument, i.e, the @client-id@, which is the client you want to send files to. We then check to be sure that the user is not trying to send files to himself. Although it isn’t possible for the user to do this, we do this as an extra layer of security. At this point, if the user does not have an active connection with the other peer, it means he is the one initiating the chat/file transfer, so we create a new connection with that @client_id@, passing the initiator status to be true. After this, we call the @render@ method to render the files that might have been sent by the other peer.
```

Comment:
Rewrite these lines:

The @prepareChat@ method receives a single argument, i.e, the @client-id@, which is the client you want to send files to. We then check to be sure that the user is not trying to send files to himself. Although it isn’t possible for the user to do this, we do this as an extra layer of security.